docs/src/main/asciidoc/kafka-guide.adoc

```diff
@@ -0,0 +1,304 @@
+== The price converter
+
+The price converter reads the prices from Kafka, and transforms them.
+Creates the `src/main/java/org/acme/quarkus/sample/PriceConverter.java` file with the following content:
+
+[source, java]
+----
+package org.acme.quarkus.sample;
+
+import io.smallrye.reactive.messaging.annotations.Broadcast;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+
+import javax.enterprise.context.ApplicationScoped;
+
+/**
+ * A bean consuming data from the "prices" Kafka topic and applying some conversion.
+ * The result is pushed to the "my-data-stream" stream which is an in-memory stream.
+ */
+@ApplicationScoped
+public class PriceConverter {
+
+    private static final double CONVERSION_RATE = 0.88;
+
+    @Incoming("prices")                                 // <1>
+    @Outgoing("my-data-stream")                         // <2>
+    @Broadcast                                          // <3>
+    public double process(int priceInUsd) {
+        return priceInUsd * CONVERSION_RATE;
+    }
+
+}
+----
+<1> Indicates that the method consumes the items from the `prices` stream
+<2> Indicates that the objects returned by the method are sent to the `my-data-stream` stream
+<3> Indicates that the item are dispatched to all _subscribers_
+
+The `process` method is called for every Kafka _record_ from the `prices` topic (configured in the application configuration).
+Every result is sent to the `my-data-stream` in-memory stream.
+
+== The price resource
+
+Finally, let's bind our stream to a Jax-RS resource.
+Creates the `src/main/java/org/acme/quarkus/sample/PriceResource.java` file with the following content:
+
+[source, java]
+----
+package org.acme.quarkus.sample;
+
+import io.smallrye.reactive.messaging.annotations.Stream;
+import org.reactivestreams.Publisher;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * A simple resource retrieving the "in-memory" "my-data-stream" and sending the items to a server sent event.
+ */
+@Path("/prices")
+public class PriceResource {
+
+    @Inject
+    @Stream("my-data-stream") Publisher<Double> prices; // <1>
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
+
+
+    @GET
+    @Path("/stream")
+    @Produces(MediaType.SERVER_SENT_EVENTS)             // <2>
+    public Publisher<Double> stream() {                 // <3>
+        return prices;
+    }
+}
+----
+<1> Injects the `my-data-stream` stream using the `@Stream` qualifier
+<2> Indicates that the content is sent using `Server Sent Events`
+<3> Returns the stream (_Reactive Stream_)
+
+== Configuring the Kafka connector
+
+We need to configure Kafka connector. This is done in the `application.properties` file.
+The keys are structured as follow:
+
+`smallrye.messaging.[sink|source].{stream-name}.property=value`
```

Comment:
We will really need to talk about if we want to achieve consistency with every configuration setting being `quarkus.<something>`. Maybe it doesn't make sense for MP components, I don't know, but I thought it was the convention.

(General comment, don't change anything)

Reply:
This is still in discussion for the spec, so likely be changed soon. But yes, the question is whether we want `quarkus.`. The implementation would support this (you just need to add a bean in the extension).